src/include/utils/rel.h

```diff
@@ -516,6 +517,23 @@ typedef struct ViewOptions
 #define RelationIsAppendOptimized(relation) \
 	AMHandlerIsAO((relation)->rd_amhandler)
 
+/*
+ * CAUTION: this macro is a violation of the absraction that table AM and
+ * index AM interfaces provide.  Use of this macro is discouraged.  If
+ * table/index AM API falls short for your use case, consider enhancing the
+ * interface.
+ *
+ * RelationIsNonblockRelation looks replacable by `!RelationIsHeap`, but
+ * they have different meanings. RelationIsNonblockRelation expand the
+ * relation type to run the code path like AO/CO. `!RelationIsHeap`
+ * emphasizes NOT heap relation.
+ *
+ * RelationIsNonblockRelation
+ *      True iff relation(table) should run the code path as AO/CO
+ */
+#define RelationIsNonblockRelation(relation) \
+	((relation)->rd_tableam && (relation)->rd_rel->relam != HEAP_TABLE_AM_OID)
+
```

Comment:
Hi, understand the change. 
 I have a idea that: do we have a certain PAX_AM_OID now? If true, it seems more reasonable to define RelationIsNonblockRelation to is PAX_AM_OID || RelationIsAppendOptimized.
Others LGTM

Reply:
PAX_AM_OID is defined in the pax code, the kernel should not depend on the oid of PAX.
We should enchance the macro if the context is changed.